sql/core/src/main/scala/org/apache/spark/sql/execution/ui/ExecutionPage.scala

```diff
@@ -75,6 +75,12 @@ class ExecutionPage(parent: SQLTab) extends WebUIPage("execution") with Logging
             {jobLinks(JobExecutionStatus.RUNNING, "Running Jobs:")}
             {jobLinks(JobExecutionStatus.SUCCEEDED, "Succeeded Jobs:")}
             {jobLinks(JobExecutionStatus.FAILED, "Failed Jobs:")}
+            <li>
+              <strong>Completed Stages: </strong>
+              {sqlStore.getStageAttempts(executionId).sorted.map { case (stage, attempt) =>
+              <a href={stageURL(request, stage, attempt)}>{stage}</a><span>&nbsp;</span>
```

Comment:
so if stage has multiple attempts, you get the same stage id multiple times in this list?  I wonder if it would be better to differentiate.... either stage(attempt) but only if there are multiple attempts or could do stage(attempt0, attempt1,...)

Reply:
Yes, though items are added to StageAttempts from `onStageCompleted` so we shouldn't be hitting this issue.